__tests__/components/AnonymousChat/child/ChatItemMyTextV2.test.tsx

```diff
@@ -1,43 +1,178 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+/* eslint-disable global-require */
+/* eslint-disable react/display-name */
 import * as React from 'react';
 import FastImage from 'react-native-fast-image';
-import {render} from '@testing-library/react-native';
+import {act, fireEvent, render} from '@testing-library/react-native';
 
 import ChatItemMyTextV2 from '../../../../src/components/AnonymousChat/child/ChatItemMyTextV2';
+import {Context} from '../../../../src/context';
 import {DEFAULT_PROFILE_PIC_PATH} from '../../../../src/utils/constants';
 
-describe('TESTING ChatItemTargetText', () => {
+jest.spyOn(React, 'useRef').mockReturnValue({current: {close: jest.fn()}});
+jest.mock('react-native-reanimated', () => ({
+  ...jest.requireActual('react-native-reanimated'),
+  useSharedValue: jest.fn(() => ({value: 0})),
+  useAnimatedStyle: jest.fn(() => ({})),
+  View: require('react-native/Libraries/Components/View/View').View,
+  Animated: {
+    View: require('react-native/Libraries/Components/View/View').View
+  }
+}));
+jest.mock('react-native-gesture-handler', () => {
+  const originalModule = jest.requireActual('react-native-gesture-handler');
+  const {View} = require('react-native');
+  const React = require('react');
+
+  return {
+    ...originalModule,
+    Swipeable: React.forwardRef(({children, ...props}, ref) => {
+      React.useImperativeHandle(ref, () => ({
+        close: jest.fn()
+      }));
+
+      return <View {...props}>{children}</View>;
+    })
+  };
+});
+
+afterEach(() => {
+  jest.restoreAllMocks();
+});
+
+describe('TESTING ChatItemMyTextV2', () => {
   const avatar = (
     <FastImage
       style={{height: 24, width: 24, borderRadius: 12}}
       source={{uri: DEFAULT_PROFILE_PIC_PATH}}
     />
   );
 
-  it('RENDER not continuous should match snapshot', () => {
-    const {toJSON} = render(
-      <ChatItemMyTextV2
-        avatar={avatar}
-        username="username"
-        isContinuous={false}
-        message="message"
-        time="time"
-      />
+  it('should call setReplyPreview when swiped', () => {
+    const setReplyPreview = jest.fn();
+    const contextValue = {
+      chat: [{replyTarget: null}, setReplyPreview],
+      profile: [{myProfile: {}}]
+    };
+
+    const {getByTestId} = render(
+      <Context.Provider value={contextValue}>
+        <ChatItemMyTextV2
+          avatar={avatar}
+          username="username"
+          isContinuous={false}
+          message="message"
+          time="time"
+          chatType="SIGNED"
+          messageType="regular"
+          data={{}}
+        />
+      </Context.Provider>
     );
 
-    expect(toJSON()).toMatchSnapshot();
+    act(() => {
+      fireEvent(getByTestId('swipeable'), 'onSwipeableOpen', 'left');
+    });
+
+    expect(setReplyPreview).toHaveBeenCalled();
   });
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This test case checks if the `setReplyPreview` function is called when swiped. However, it does not check what arguments are passed to the function. It would be more robust to also check the arguments to ensure the correct data is being passed.



<!-- This is an auto-generated comment by OSS CodeRabbit -->